CMakeLists.txt

```diff
@@ -73,54 +33,175 @@ if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+	# Use system package
+	find_lib_req(liblcf ON)
 endif()
 
-# SDL_mixer
-find_package(SDL2_mixer)
-if(SDL2_MIXER_FOUND)
-  include_directories(${SDL2_MIXER_INCLUDE_DIR})
-  list(APPEND EASYRPG_PLAYER_LIBRARIES ${SDL2_MIXER_LIBRARY})
-  add_definitions(-D HAVE_SDL_MIXER=1)
+# Detect all required libraries
+foreach(i Pixman ZLIB PNG SDL2)
+	find_lib_req(${i} ON)
+endforeach()
+
+# Always enable Wine registry support on non-Windows
+if(CMAKE_SYSTEM_NAME MATCHES "Windows")
 else()
-  add_definitions(-DNO_SDL_MIXER)
+	add_definitions(-DHAVE_WINE=1)
 endif()
 
-# OpenAL
-find_package(OpenAL)
-if(OPENAL_FOUND)
-  find_package(SndFile REQUIRED)
-  find_package(FluidSynth REQUIRED)
-  include_directories(${OPENAL_INCLUDE_DIR} ${SNDFILE_INCLUDE_DIR} ${FLUIDSYNTH_INCLUDE_DIR})
-  list(APPEND EASYRPG_PLAYER_LIBRARIES ${OPENAL_LIBRARY} ${SNDFILE_LIBRARIES} ${FLUIDSYNTH_LIBRARY})
-  add_definitions(-D HAVE_OPENAL=1)
+# freetype and harfbuzz
+option(PLAYER_WITH_FREETYPE "Support FreeType font rendering" ON)
+find_lib("Freetype" PLAYER_WITH_FREETYPE)
+if(${RETVAL})
+	add_definitions(-DHAVE_FREETYPE)
+	option(PLAYER_WITH_HARFBUZZ "Enable HarfBuzz text shaping (Requires FreeType)" ON)
+	find_lib("Harfbuzz" PLAYER_WITH_HARFBUZZ)
+	if(${RETVAL})
+		add_definitions(-DHAVE_HARFBUZZ)
+	endif()
 endif()
 
-include_directories("${CMAKE_CURRENT_SOURCE_DIR}/lib/liblcf/src")
-include_directories("${CMAKE_CURRENT_SOURCE_DIR}/lib/liblcf/src/generated")
-include_directories("${CMAKE_CURRENT_SOURCE_DIR}/src")
-
-foreach(i Expat Freetype Harfbuzz Pixman ZLIB PNG SDL2 Iconv)
-  find_package(${i} REQUIRED)
-
-  string(TOUPPER ${i} i)
+# Sound system to use
+set(PLAYER_AUDIO_BACKEND "SDL2_mixer" CACHE STRING "Audio system to use. The SDL2_mixer audio system provides advanced features required by RPG Maker. Options: SDL2_mixer OpenAL OFF")
+set_property(CACHE PLAYER_AUDIO_BACKEND PROPERTY STRINGS SDL2 OpenAL)
+
+# Configure Audio backends
+
+if(${PLAYER_AUDIO_BACKEND} MATCHES "SDL2_mixer")
+	# SDL2_mixer Audio
+	find_lib_req(SDL2_mixer ON)
+	add_definitions(-D HAVE_SDL_MIXER=1)
+	
+	# speexdsp
+	option(PLAYER_WITH_SPEEXDSP "Resample audio with speexdsp." ON)	
+	find_lib("speexdsp" PLAYER_WITH_SPEEXDSP)
+	if(${RETVAL})
+		add_definitions(-DHAVE_LIBSPEEXDSP)
+	endif()
+	
+	# mpg123
+	option(PLAYER_WITH_MPG123 "Play MP3 audio with libmpg123." ON)
+	find_lib("mpg123" PLAYER_WITH_MPG123)
+	if(${RETVAL})
+		add_definitions(-DHAVE_MPG123)
+	endif()
+
+	# libsndfile
+	option(PLAYER_WITH_LIBSNDFILE "Play WAV audio with libsndfile." ON)	
+	find_lib("SndFile" PLAYER_WITH_LIBSNDFILE)
+	if(${RETVAL})
+		add_definitions(-DHAVE_LIBSNDFILE)
+	endif()
+	
+	# libogg
+	option(PLAYER_WITH_LIBOGG "Play OGG audio with libogg or tremor." ON)	
+	find_lib("OggVorbis" PLAYER_WITH_SPEEXDSP)
+	if(${RETVAL})
+		add_definitions(-DHAVE_OGGVORBIS)
+	endif()
+	
+	# wildmidi (optional, OFF)
+	option(PLAYER_WITH_WILDMIDI "Play MIDI audio with wildmidi. (optional)" OFF)
+	find_lib("wildmidi" PLAYER_WITH_WILDMIDI)
```

Comment:
This works because find_library uses that argument to open "Findwildmidi.cmake" but I can rename both to be consistent with the lib name (WildMidi)